aria/cli/core/aria.py

```diff
@@ -65,15 +58,33 @@ def __init__(self, *args, **kwargs):
         super(MutuallyExclusiveOption, self).__init__(*args, **kwargs)
 
     def handle_parse_result(self, ctx, opts, args):
-        if self.mutually_exclusive.intersection(opts) and self.name in opts:
+        if (self.name in opts) and self.mutually_exclusive.keys().intersection(opts):
             raise click.UsageError(
                 'Illegal usage: `{0}` is mutually exclusive with '
                 'arguments: [{1}] ({2}).'.format(
                     self.name,
                     self.mutuality_string,
                     self.mutuality_error_message))
-        return super(MutuallyExclusiveOption, self).handle_parse_result(
-            ctx, opts, args)
+        return super(MutuallyExclusiveOption, self).handle_parse_result(ctx, opts, args)
+
+
+def mutually_exclusive_option(*param_decls, **attrs):
+    """
+    Makes options mutually exclusive. The decorator must pass a a ``mutually_exclusive`` argument
+    with a list of argument names with which the option is mutually exclusive.
+
+    NOTE: All mutually exclusive options must use this. It's not enough to use it in just one of the
+    options.
+    """
+    def decorator(func):
+        if 'help' in attrs:
```

Comment:
could you please document this func a bit more?
I'm not sure what are `param_decls` and `attrs` in this case, why `help` is being inspected, and why the internal `__click_params__` is accessed directly here.